File: Ruby/selection_sort.rb
def selection_sort(array)
    for i in (0..array.length)
        min = i
        for j in (i+1..array.length-1)
            if array[j] < array[min]
                min = j
            end
        end
        if i != min
            array[i], array[min] = array[min], array[i]
        end
    end
    return array
end

puts selection_sort([5, 4, 3, 2, 1])

File: Python/Algorithm/selection_sort.py
def selection_sort(li: list[int]) -> list[int]:
    for i in range(0, len(li)):
        min = i

        for j in range(i + 1, len(li)):
            if li[min] > li[j]:
                min = j

        if i != min:
            temp = li[i]
            li[i] = li[min]
            li[min] = temp

    return li


li = [5, 4, 3, 2, 1]
sorted_array = selection_sort(li)

print(sorted_array)
